pkg/storage/fs/owncloud/upload.go

```diff
@@ -375,6 +383,54 @@ func (upload *fileUpload) FinishUpload(ctx context.Context) error {
 			}
 		}
 	*/
+	log := appctx.GetLogger(upload.ctx)
+
+	sha1h := make([]byte, 0, 32)
+	md5h := make([]byte, 0, 32)
+	adler32h := make([]byte, 0, 32)
+	{
+		sh := sha1.New()
+		mh := md5.New()
+		ah := adler32.New()
+		f, err := os.Open(upload.binPath)
+		if err != nil {
+			log.Err(err).Msg("Decomposedfs: could not open file for checksumming")
+			// we can continue if no oc checksum header is set
+		}
+		defer f.Close()
+
+		r1 := io.TeeReader(f, sh)
+		r2 := io.TeeReader(r1, mh)
+
+		if _, err := io.Copy(ah, r2); err != nil {
+			log.Err(err).Msg("Decomposedfs: could not copy bytes for checksumming")
+		}
+
+		sha1h = sh.Sum(sha1h)
+		md5h = mh.Sum(md5h)
+		adler32h = ah.Sum(adler32h)
+	}
+
+	if upload.info.MetaData["checksum"] != "" {
+		parts := strings.SplitN(upload.info.MetaData["checksum"], " ", 2)
+		if len(parts) != 2 {
+			return errtypes.BadRequest("invalid checksum format. must be '[algorithm] [checksum]'")
+		}
+		var err error
+		switch parts[0] {
+		case "sha1":
+			err = upload.checkHash(parts[1], sha1h)
+		case "md5":
+			err = upload.checkHash(parts[1], md5h)
+		case "adler32":
+			err = upload.checkHash(parts[1], adler32h)
+		default:
+			err = errtypes.BadRequest("unsupported checksum algorithm: " + parts[0])
+		}
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
This means that we do not require checksums, right? But if  the feature "checksums" is activated, shouldn't it be an error/warning if the upload comes completely without checksum? Or is that checked elsewhere?

Reply:
Yes, as it is right now checksums are optional. I don't know if we even have the flag to "activate" checksums.
Also I'm not sure how much sense it makes to spend too much effort in the owncloud storage driver since we are only using it for the webUI acceptance tests and even them will in the near future need to switch to the ocis storage driver.

This PR is just to make the webUI tests for the Thumbnails PR pass. https://github.com/owncloud/ocis/pull/1898